Implement Association Browser screen with modular components and localised strings

KianBahia opened the pull request:
This PR implements the **Association Browser** screen and improves overall app maintainability by externalising hardcoded text and adding Figma design documentation.
It also addresses issues #39, #55, and #57.

---

### Association Browser Screen #39 
- Implemented a new **`AssociationBrowser`** composable screen.
- Includes:
  - Search bar aligned with the Home screen layout (with logo spacing preserved).
  - `DisplayedAssociationFilter` component for toggling between **Subscribed** and **All Associations**.
  - `AssociationCard` component for clean, reusable association display.
  - Empty state message when the user has no subscribed associations.
  - Added a temporary placeholder image (R.drawable.placeholder) for association logos until real profile images are supported.

### Modular Components
- Added **`AssociationCard`** composable — mirrors the `EventCard` structure for consistency.
- Created **`DisplayedAssociationFilter`** — matches the filter style used in the Home screen.

###  Strings Externalisation
- Removed all hardcoded UI text from `HomeScreen` and `AssociationBrowser` to comply with issue #57  
- Moved all text into `res/values/strings.xml` for localisation and easier maintenance.
- Examples include:
  - “Subscribed”
  - “All Associations”
  - “There’s nothing here!”
  - “No subscribed associations yet!”

### Figma Design Documentation #55 
- Added new **high-fidelity mockup screenshots** and **Mockup User Flows** under `figma-work/mockup-screenshots/`.
- Updated `docs/figma-designs.md` to include:
  - A new **Mockup Designs** section.
  - Embedded images for all major app screens and user flows.
  - Captions and layout matching the wireframes and user flow sections.

---

## Files Added or Updated

| File | Description |
|------|--------------|
| `ui/association/AssociationBrowser.kt` | New Association Browser screen composable |
| `ui/composables/AssociationCard.kt` | Card component for association display |
| `ui/composables/DisplayedAssociationFilter.kt` | Filter component for Subscribed/All views |
| `res/values/strings.xml` | Updated to add all text from Home and AssociationBrowser |
| `docs/figma-designs.md` | Updated with new mockup screenshots section |
| `figma-work/mockup-screenshots/*` | Added Figma mockup images |

---

## Design & UX Reference

All visual and layout decisions are based on the **Figma design system** and verified against the new mockup screenshots added in this PR.

---

## Screenshots below
<img width="311" height="680" alt="image" src="https://github.com/user-attachments/assets/bf328eb2-0e74-4ca0-9380-f51640962a2c" />
<img width="307" height="672" alt="image" src="https://github.com/user-attachments/assets/5e4ffdab-1d13-45fc-8248-0c6112bb2d8f" />
<img width="1177" height="945" alt="AllMockups" src="https://github.com/user-attachments/assets/7ef2e157-df51-489b-9e79-276e8ca1c72d" />

KianBahia committed:
HomeScreen shows message when no subscribed events are available

Adds EmptyEventsMessage composable to display a hint for new users. Updates EventCard styling to improve dropshadow.

# Conflicts:
#	app/src/main/java/ch/epfllife/ui/composables/EventCard.kt
#	app/src/main/java/ch/epfllife/ui/home/HomeScreen.kt

KianBahia committed:
Apply ktfmt formatting fixes

KianBahia committed:
Implement Association Browser screen and externalize UI strings

- Added full Association Browser screen with modular AssociationCard and filter components
- Added empty state messages for unsubscribed users
- Moved all hardcoded text from HomeScreen and AssociationBrowser into strings.xml
- Updated filters to use localized string resources for labels

Adriv020 reviewed (commented):
Great work on this PR, the Association Browser screen looks really good and integrates seamlessly with the rest of the app.
I also wanted to thank you for the work on the Figma mockups, they look excellent.

That said, I noticed that the first two commits (HomeScreen shows message when no subscribed events are available and Apply ktfmt formatting fixes) are duplicates from another PR #56 .
They probably got reintroduced by mistake due to a rebase or branch merge. It’s not a big issue, but it’s good to clean that up next time to keep the history clear.

One small suggestion for future PRs: try to split large changes into smaller, logical commits (for example: one for the new screen, one for UI strings, one for documentation).
This makes reviews easier and helps other contributors follow the evolution of the feature.

Please also take a look at the comments I left in the code review.

Adriv020 commented on app/src/main/java/ch/epfllife/ui/composables/DisplayedAssociationFilter.kt:
```diff
@@ -0,0 +1,69 @@
+import ch.epfllife.R
+
+@Composable
+fun DisplayedAssociationFilter(
```
I noticed that both DisplayedAssociationFilter and EventsFilterButtons are almost identical, sharing the same structure, layout, and logic, only differing in the enum and label texts. 
It might be worth considering an approach using inheritance or an abstract base composable, where both filters could extend a shared implementation that handles the common layout logic (row, text style, divider, etc.).

This would reduce duplicated code while still allowing each implementation to define its own enum or specific labels.

KianBahia committed:
Remove separate filter classes from model.enum to make a single SubscriptionFilter class

KianBahia committed:
refactor(ui): unify event and association filters into DisplayedSubscriptionFilter

- Removed duplicate DisplayedAssociationFilter composable
- Renamed EventsFilterButtons to DisplayedSubscriptionFilter
- Updated HomeScreen and AssociationBrowser to use the new shared component

KianBahia committed:
Handle empty “All” states for Events and Associations

- Added separate empty state messages for when there are no events or associations at all
- Refactored EmptyEventsMessage and EmptyAssociationsMessage to accept dynamic title/description
- Updated strings.xml with new message strings for these cases

Adriv020 commented on app/src/main/java/ch/epfllife/ui/composables/DisplayedSubscriptionFilter.kt:
```diff
@@ -0,0 +1,75 @@
+import ch.epfllife.model.enums.SubscriptionFilter
+
+@Composable
+fun DisplayedSubscriptionFilter(
```
Perfect, with this solution we abstract the two existing Filter composables into a single one.

KianBahia committed:
Apply ktfmt auto-formatting to fix style check

KianBahia committed:
Merge branch 'main' into association-browser

KianBahia committed:
Handle empty “All” states for Events and Associations

- Added separate empty state messages for when there are no events or associations at all
- Refactored EmptyEventsMessage and EmptyAssociationsMessage to accept dynamic title/description
- Updated strings.xml with new message strings for these cases

KianBahia committed:
Merge branch 'main' into association-browser

KianBahia committed:
Refactored Events Filter test

KianBahia committed:
Merge remote-tracking branch 'origin' into association-browser

KianBahia committed:
format

AlecGhost commented:
Closing this PR since it got replaced by #95 due to merging problems.
